base/compiler/inferenceresult.jl

```diff
@@ -13,32 +13,27 @@ mutable struct InferenceResult
         else
             result = linfo.rettype
         end
-        return new(linfo, EMPTY_VECTOR, result, nothing)
+        return new(linfo, compute_inf_result_argtypes(linfo), result, nothing)
```

Comment:
Originally, I was going to basically rename `get_argtypes(::InferenceResult)` to `init_argtypes`,  but then I couldn't find a place where it ever looked like we wanted to construct an InferenceResult without initializing the `argtypes`. Hence this change, and getting rid of `get_argtypes(::InferenceResult)` altogether. 

Am I correct here that - currently, anyway - all cases where we construct an `InferenceResult`, we also want to initialize `argtypes`?